components/search/src/filter.rs

```diff
@@ -213,14 +258,23 @@ impl Filter for Vec<JSONSearchConfigurationRecords> {
         user_environment: &SearchUserEnvironment,
         overrides: Option<Vec<JSONOverridesRecord>>,
     ) -> Result<FilterRecordsResult, Error> {
+        let mut available_locales = Vec::new();
+        for record in self {
+            if let JSONSearchConfigurationRecords::AvailableLocales(locales_record) = record {
+                available_locales = locales_record.locales.clone();
+            }
+        }
+        let mut user_environment = user_environment.clone();
```

Comment:
I think we should pass in a cloned, mutable user_environment here - currently I think `filter_engine_configuration_impl` would not get the updated locale, and hence it might determine the sort order/default engines incorrectly.